### ISSUE TITLE
Fill in autogenerated correlationId if sync publish

### ISSUE THREAD

**Issue by llchan** [liftbridge-io/liftbridge#122]
I think the current implementation of the sync Publish may listen on a shared ackInbox without a correlationId.  It should always force a non-empty correlationId and verify it when the acks come in.  For the fire-and-forget, that verification is let to the caller, which I think is reasonable, but if we're internally blocking and waiting we should make sure we wait for the right one.

**Comment by tylertreat** [comment]
Yes, you're right. Currently, we [assign a unique ack inbox](https://github.com/liftbridge-io/liftbridge/blob/88395aee8dfa55ae585fc1570f9b2a7f66409f33/server/api.go#L142-L144) if one is not provided, but we're currently not verifying the correlation id if an ack inbox _is_ provided/re-used by the client.
